pkg/nuclide/ui/tree/lib/TreeRootComponent.js

```diff
@@ -178,6 +190,35 @@ var TreeRootComponent = React.createClass({
     this.setState({expandedKeys});
   },
 
+  async _toggleChildNodesExpanded(parentNode: LazyTreeNode, forceExpanded: ?boolean): Promise {
```

Comment:
That seems like a better approach because those methods already implement expanding/collapsing. They could be recursively expand children of the node rather than implement that in a separate function.